File: library/src/main/java/com/sideslip/view/ItemBind.java
package com.sideslip.view;

import com.sideslip.face.IItemBind;

/**
 * created by wyu on 2021/3/6.
 */


public abstract class ItemBind<T> implements IItemBind {


    @Override
    public final void bind(ItemView itemView, Object data, int position) {
        onBind(itemView, (T) data, position);
    }


    public abstract void onBind(ItemView itemView, T data, int position);
}

File: library/src/main/java/com/sideslip/view/ItemBind.kt
package com.sideslip.view

import com.sideslip.face.IItemBind

/**
 * created by wyu on 2021/3/6.
 */
abstract class ItemBind<T> : IItemBind {
    override fun bind(itemView: ItemView, data: Any, position: Int) {
        onBind(itemView, data as T, position)
    }

    abstract fun onBind(itemView: ItemView, data: T, position: Int)
}